PhysicsTools/NanoAOD/python/l1trig_cff.py

```diff
@@ -0,0 +1,119 @@
+    cut = cms.string(""), 
+    name= cms.string("L1Jet"),
+    doc = cms.string(""),
+    extension = cms.bool(False), # this is the main table for L1 EGs
+    variables = cms.PSet(l1ObjVars,
+                         towerIEta = Var("towerIEta()",int,doc=""),
+                         towerIPhi = Var("towerIPhi()",int,doc=""),
+                         rawEt = Var("rawEt()",int,doc=""),
+                         seedEt = Var("seedEt()",int,doc=""),
+                         puEt = Var("puEt()",int,doc=""),
+                         puDonutEt0 = Var("puDonutEt(0)",int,doc=""),
+                         puDonutEt1 = Var("puDonutEt(1)",int,doc=""),
+                         puDonutEt2 = Var("puDonutEt(2)",int,doc=""),
+                         puDonutEt3 = Var("puDonutEt(3)",int,doc=""),
+                     )
+)
+
+l1TauTable = cms.EDProducer("SimpleTriggerL1TauFlatTableProducer",
+    src = cms.InputTag("caloStage2Digis","Tau"),
+    minBX = cms.int32(-2),
+    maxBX = cms.int32(2),                           
+    cut = cms.string(""), 
+    name= cms.string("L1Tau"),
+    doc = cms.string(""),
+    extension = cms.bool(False), # this is the main table for L1 EGs
+    variables = cms.PSet(l1ObjVars,
+                         towerIEta = Var("towerIEta()",int,doc=""),
+                         towerIPhi = Var("towerIPhi()",int,doc=""),
+                         rawEt = Var("rawEt()",int,doc=""),
+                         isoEt = Var("isoEt()",int,doc=""),
+                         nTT = Var("nTT()",int,doc=""),                         
+                         hasEM = Var("hasEM()",int,doc=""),
+                         isMerged = Var("isMerged()",int,doc=""),
+
+                     )
+)
+
+l1EtSumTable = cms.EDProducer("SimpleTriggerL1EtSumFlatTableProducer",
+    src = cms.InputTag("caloStage2Digis","EtSum"),
+    minBX = cms.int32(-2),
+    maxBX = cms.int32(2),                           
+    cut = cms.string(""), 
+    name= cms.string("L1EtSum"),
+    doc = cms.string(""),
+    extension = cms.bool(False), # this is the main table for L1 EGs
+    variables = cms.PSet(PTVars,
+                         hwPt = Var("hwPt()",int,doc="hardware pt"),
+                         hwPhi = Var("hwPhi()",int,doc="hardware phi"),
+                         etSumType = Var("getType()",int,doc=""),
+                     )
+)
+
+l1EGTable = cms.EDProducer("SimpleTriggerL1EGFlatTableProducer",
+    src = cms.InputTag("caloStage2Digis","EGamma"),
+    minBX = cms.int32(-2),
+    maxBX = cms.int32(2),                           
+    cut = cms.string(""), 
+    name= cms.string("L1EG"),
+    doc = cms.string(""),
+    extension = cms.bool(False), # this is the main table for L1 EGs
+    variables = cms.PSet(l1ObjVars,
+                         towerIEta = Var("towerIEta()",int,doc="tower ieta"),
+                         towerIPhi = Var("towerIPhi()",int,doc="tower iphi"),
+                         rawEt = Var("rawEt()",int,doc="raw et"),
```

Comment:
`towerIEta`, `towerIPhi`, and `rawEt` are repeated for `l1EG`'s, `l1Tau`'s and `l1Jet`'s. You could maybe save a few lines by just making those a PSet too

Reply:
thanks for the comment, we've now reworked this file a little and will consider if it still makes sense to do this which it probably does as it means we only have to write the doc string oncey